mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -345,6 +302,31 @@ def after_create(self, block, **kwargs):
                 widget=widget,
             )
 
+    @classmethod
+    def block_class_from_type(self, block_type: str, language=None, pipeline=None) -> 'Block':
+        from mage_ai.data_preparation.models.block.constants import BLOCK_TYPE_TO_CLASS
+        from mage_ai.data_preparation.models.block.integration import (
+            SourceBlock, DestinationBlock, TransformerBlock
+        )
+        from mage_ai.data_preparation.models.block.r import RBlock
+        from mage_ai.data_preparation.models.block.sql import SQLBlock
+        from mage_ai.data_preparation.models.widget import Widget
+
+        if block_type == BlockType.CHART:
+            return Widget
+        if pipeline and PipelineType.INTEGRATION == pipeline.type:
+            if BlockType.DATA_LOADER == block_type:
+                return SourceBlock
+            elif BlockType.DATA_EXPORTER == block_type:
+                return DestinationBlock
+            else:
+                return TransformerBlock
+        elif BlockLanguage.SQL == language:
```

Comment:
The BlockLanguage for DBT block is also SQL. Did you test DBT blocks?

Reply:
hmm I did test DBT blocks, but let me double check